### ISSUE TITLE
Version 2.3.2 of vpn_ha introduces new breaking constraints

### ISSUE THREAD

**Issue by mattatg** [terraform-google-modules/terraform-google-vpn#146]
### TL;DR

Tag 2.3.2 of the vpn_ha module, introduces a new constraint of terraform < 1.3. This breaks anyone that depends on `~> 2.3.1` of this module.

### Expected behavior

Patch releases should not introduce unnecessary breaking changes.  v2.3.3 should retract this terraform constraint, allowing callers to run more modern versions of terraform while still pinning to this major+minor version of the modules (v2.3.*)

### Observed behavior

Running `terraform init` with terraform cli > 1.3, and reference to this module version ~> 2.3.1  returns the following error:

```
│ Error: Unsupported Terraform Core version
│
│   on .terraform//modules/vpn_ha/versions.tf line 18, in terraform:
│   18:   required_version = ">=0.13.0, < 1.3"
│
│ Module module.this (from registry.terraform.io/terraform-google-modules/vpn/google//modules/vpn_ha) does not support Terraform version 1.4.6. To proceed, either choose another supported Terraform version or update this version
│ constraint. Version constraints are normally set for good reason, so updating the constraint may lead to other errors or unexpected behavior.
```

### Terraform Configuration

```hcl
module "this" {
  source = "terraform-google-modules/vpn/google//modules/vpn_ha"
  version = "~> 2.3.1"
```
```


### Terraform Version

```sh
1.4.6
```

### Additional information

This can be resolved by pinning directly to 2.3.1 or any version greater than 2.3.2, but it breaks the whole point of patch versioning and should be fixed (despite being a simple fix)

**Comment by adriantr** [comment]
+1 on this one. have a PR [here](https://github.com/GoogleCloudPlatform/terraform-google-secure-cicd/pull/170) to avoid 2.3.2 for cloudbuild networking, but I assume it has a broader impact.

**Comment by Gipphe** [comment]
`v2.3.3` fixes the version constraint, but there are other issues causing `v2.3.2` and thus `v2.3.3` to be breaking changes, such as [`bgp-session-name`](https://github.com/terraform-google-modules/terraform-google-vpn/compare/v2.3.1...v2.3.2#diff-aa8412dfa1d9d1daa4185fcb16c825cd4a58a61e307d43686dbc9c5bd6fa5f7aR103), [`peed_external_gateway_self-link`](https://github.com/terraform-google-modules/terraform-google-vpn/compare/v2.3.1...v2.3.2#diff-aa8412dfa1d9d1daa4185fcb16c825cd4a58a61e307d43686dbc9c5bd6fa5f7aR114) and [`ip_address`](https://github.com/terraform-google-modules/terraform-google-vpn/compare/v2.3.1...v2.3.2#diff-aa8412dfa1d9d1daa4185fcb16c825cd4a58a61e307d43686dbc9c5bd6fa5f7aR105) are now required properties in `tunnels`.

**Comment by imrannayer** [comment]
in [Change log](https://github.com/terraform-google-modules/terraform-google-vpn/blob/master/CHANGELOG.md) it shows 2.4 after 2.3.1. 
Strange thing is [Registry](https://registry.terraform.io/modules/terraform-google-modules/vpn/google/4.0.0) is showing 2.3.2 and 2.3.3 release.

**Comment by adriantr** [comment]
@g-awmalik bump
